File: app/src/main/java/com/kite/mnemoai/ui/vocabularybookgroup/VocabularyGroupUIState.java
package com.kite.mnemoai.ui.vocabularybookgroup;

import com.kite.mnemoai.data.local.entity.GroupEntity;
import com.kite.mnemoai.data.model.WordListItem;

import java.util.List;

public class VocabularyGroupUIState {
    private GroupEntity group;
    private List<WordListItem> words;

    public VocabularyGroupUIState(List<WordListItem> words) {
        this.words = words;
    }

    public VocabularyGroupUIState(GroupEntity group, List<WordListItem> words) {
        this.group = group;
        this.words = words;
    }

    public List<WordListItem> getWords() {
        return words;
    }

    public VocabularyGroupUIState(GroupEntity group) {
        this.group = group;
    }

    public GroupEntity getGroup() {
        return group;
    }
}

File: feature/vocabularybookgroup/src/main/java/com/kite/mnemoai/vocabularybookgroup/VocabularyGroupUIState.kt
package com.kite.mnemoai.vocabularybookgroup

import com.kite.mnemoai.model.group.Group
import com.kite.mnemoai.model.word.WordItem

data class VocabularyGroupUIState(
    val group: Group?,
    val words: List<WordItem>?,
    val allCount: Long = 0,
    val learningCount: Long = 0,
    val reviewingCount: Long = 0,
    val masteredCount: Long = 0,
)